_pytest/capture.py

```diff
@@ -252,6 +252,8 @@ def writelines(self, linelist):
         self.write(data)
 
     def __getattr__(self, name):
+        if name == 'name':
+            return repr(self.buffer)
         return getattr(object.__getattribute__(self, "buffer"), name)
```

Comment:
OH that! Hmm yeah at first glance you are right, but I suspect this is there for a reason. We would have to hunt the Git history to see if we can find something about it.